README.md

```diff
@@ -655,25 +655,93 @@ _You can enable the following settings in Xcode by running [this script](resourc
 
   </details>
 
-* <a id='attributes-on-prev-line'></a>(<a href='#attributes-on-prev-line'>link</a>) **Place function/type attributes on the line above the declaration**. [![SwiftFormat: wrapAttributes](https://img.shields.io/badge/SwiftFormat-wrapAttributes-7B0051.svg)](https://github.com/nicklockwood/SwiftFormat/blob/master/Rules.md#wrapAttributes)
+* <a id='attributes-on-prev-line'></a>(<a href='#attributes-on-prev-line'>link</a>) **Place attributes for functions, types, and computed properties on the line above the declaration**. [![SwiftFormat: wrapAttributes](https://img.shields.io/badge/SwiftFormat-wrapAttributes-7B0051.svg)](https://github.com/nicklockwood/SwiftFormat/blob/master/Rules.md#wrapAttributes)
 
   <details>
 
   ```swift
   // WRONG
   @objc class Spaceship {
 
+    @ViewBuilder var controlPanel: some View {
+      // ...
+    }
+
     @discardableResult func fly() -> Bool {
+      // ...
     }
+
   }
 
   // RIGHT
   @objc
   class Spaceship {
 
+    @ViewBuilder
+    var controlPanel: some View {
+      // ...
+    }
+
     @discardableResult
     func fly() -> Bool {
+      // ...
     }
+
+  }
+  ```
+
+  </details>
+
+* <a id='stored-property-attributes-on-same-line'></a>(<a href='#stored-property-attributes-on-same-line'>link</a>) **Place simple attributes for stored properties on the same line as the rest of the declaration**. Complex attributes with arguments should be placed on the previous line. Despite having an argument, the SwiftUI `@Environment` attribute should be placed on the same line as the rest of the declaration. [![SwiftFormat: wrapAttributes](https://img.shields.io/badge/SwiftFormat-wrapAttributes-7B0051.svg)](https://github.com/nicklockwood/SwiftFormat/blob/master/Rules.md#wrapAttributes)
+
+  <details>
+
+  ```swift
+  // WRONG
+  struct SpaceshipDashboardView {
+
+    @State
+    private var warpDriveEnabled: Bool
+
+    @ObservedObject
+    private var lifeSupportService: LifeSupportService
+
+    @Environment(\.controlPanelStyle) 
+    private var controlPanelStyle
+
+  }
+
+  // RIGHT
+  struct SpaceshipDashboardView {
+
+    @State private var warpDriveEnabled: Bool
+
+    @ObservedObject private var lifeSupportService: LifeSupportService
+
+    @Environment(\.controlPanelStyle) private var controlPanelStyle
+
+  }
+  ```
+
+  ```
+  // WRONG
+  struct RocketFactory {
+
+    @available(*, unavailable, message: "No longer in production") var saturn5Builder: Saturn5Builder
+
+    @available(*, deprecated, message: "To be retired by 2030") var atlas5Builder: Atlas5Builder
+
+  }
+
+  // RIGHT
+  struct RocketFactory {
```

Comment:
We talked outside of this PR about the different types of attributes that we are aware of:

- Property wrappers
- Special Swift language / Apple attributes (@objc, @discardableResult, @available, etc)
- Attached macros
- Result builders

I think it would be nice if our code examples had more broad coverage of these different types of attributes, for completeness.

For example, I think we should have an example of an attached macro. Additionally, I would find it helpful to have an example of a non-SwiftUI property wrapper, though I feel less strongly about this one.

Reply:
We have examples of all of these except attached macros. All of these examples so far represent real-world use cases pretty well, and I currently don't know of any real-world use cases of attached macros on stored properties (outside of hypothetical potential use cases that we could come up with). Can you think of any?

I'd maybe prefer to not contrive a hypothetical use case just for the sake of including an example.

Reply:
> All of these examples so far represent real-world use cases pretty well, and I currently don't know of any real-world use cases of attached macros on stored properties (outside of hypothetical potential use cases that we could come up with). Can you think of any?

@calda I haven't used this macro myself, but what do you think about the [`@Query` macro from SwiftData](https://developer.apple.com/documentation/swiftdata/preserving-your-apps-model-data-across-launches#Fetch-models-for-display-or-additional-processing)?

Reply:
Great suggestion, thanks! I'll add it.

Reply:
Here you go https://github.com/airbnb/swift/pull/254/commits/ad59cbe186b4a54eba44e572c00344aa899e7295